rand_distr/src/zipf.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright 2021 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The Zeta and related distributions.
+
+use num_traits::Float;
+use crate::{Distribution, Standard};
+use rand::{Rng, distributions::OpenClosed01};
+use core::fmt;
+
+/// Samples floating-point numbers according to the zeta distribution.
+///
+/// The zeta distribution is a limit of the [`Zipf`] distribution.
```

Comment:
I looked at [Geometric](https://docs.rs/rand_distr/0.4.1/rand_distr/struct.Geometric.html) and there is information about the actual support of the implementation, I think we should add it here too. 

Note: The distribution is actually over integers (more later).

Reply:
I clarified in the docs that they are over integers. I still would like to leave the return type as `f64`, see below.

Reply:
Okay!

Reply:
@saona-raimundo you can mark this as resolved?

Reply:
@dhardy Sorry, I don't see the `Resolve` button. Although I accepted the invitation you sent me. (I am not sure how to check this)